lib/stitches/cache_injector.rb

```diff
@@ -0,0 +1,32 @@
+require 'lru_redux'
+
+module Stitches::CacheInjector
+  MAX_ITEMS_IN_CACHE = ENV.fetch("API_KEY_CACHE_MAX_ITEMS", 100)
+  CACHE_DURATION_IN_SECONDS = ENV.fetch("API_KEY_CACHE_DURATION", 5 * 60) # five minutes
+
+  def self.inject
+    ::ApiClient.define_singleton_method 'api_key_cache' do
```

Comment:
Are there benefits to defining `.api_key_cache` and other methods dynamically, versus just adding them to the `ApiClient` class?  In terms of readability, IMO having the methods live inside `ApiClient` would be the more discoverable, but I'm guessing it's due to the difference between new apps that use the generators, and existing apps that have already generated the `ApiClient` class?

Reply:
I think I'm going to follow up on @samsm's idea for the API above, but the reason I was doing this dynamically, is that stitches relies on an ApiClient model to exist in the calling code base, not in this codebase.